src/components/Environment/ApplicationEnvironmentCards.tsx

```diff
@@ -59,21 +65,29 @@ const ApplicationEnvironmentCard: React.FC<ApplicationEnvironmentCardProps> = ({
 
 type ApplicationEnvironmentCardsProps = {
   isExpanded: boolean;
+  onSelect?: (selectedId: string) => void;
 };
 
 export const ApplicationEnvironmentCards: React.FC<ApplicationEnvironmentCardsProps> = ({
   isExpanded,
+  onSelect,
 }) => {
   const [environments, loaded] = useEnvironments();
   return (
-    <>
-      {loaded && environments?.length > 0
-        ? environments.slice(0, 2).map((env) => (
-            <FlexItem key={env.metadata.uid}>
-              <ApplicationEnvironmentCard environment={env} isExpanded={isExpanded} />
-            </FlexItem>
-          ))
-        : null}
-    </>
+    <Flex alignItems={{ default: 'alignItemsCenter' }}>
+      {loaded && environments?.length > 0 ? (
+        environments.slice(0, 2).map((env) => (
+          <FlexItem key={env.metadata.uid}>
+            <ApplicationEnvironmentCard
+              environment={env}
+              isExpanded={isExpanded}
+              onSelect={() => onSelect && onSelect(env.metadata.name)}
+            />
+          </FlexItem>
+        ))
+      ) : (
+        <Spinner isSVG size="md" />
```

Comment:
do we have an empty state for applications with no environments? spinner looks it is still loading..